src-java/stats-topology/stats-storm-topology/src/test/java/org/openkilda/wfm/topology/stats/StatsTopologyTest.java

```diff
@@ -426,6 +513,715 @@ public void flowStatsTest() {
+        validateHaFlowStats(HA_FLOW_ID_1, reverseEgress, cookieReverse, SWITCH_ID_1, false,
+                true, false, SUB_FLOW_ID_SHARED);
+
+        FlowStatsEntry reverseIngressPoint2 = new FlowStatsEntry(1, COOKIE_REVERSE_SUBFLOW_2.getValue(),
+                60L, 190L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_6, Collections.singletonList(reverseIngressPoint2)));
+        validateHaFlowStats(HA_FLOW_ID_1, reverseIngressPoint2, COOKIE_REVERSE_SUBFLOW_2, SWITCH_ID_6,
+                true, false, false, SUB_FLOW_ID_2);
+
+        FlowStatsEntry reverseYPoint2 = new FlowStatsEntry(1, COOKIE_REVERSE_SUBFLOW_2.getValue(),
+                50L, 180L, 10, 10);
+        sendStatsMessage(new FlowStatsData(SWITCH_ID_3, Collections.singletonList(reverseYPoint2)));
+        validateHaFlowStats(HA_FLOW_ID_1, reverseYPoint2, COOKIE_REVERSE_SUBFLOW_2, SWITCH_ID_3, false,
+                false, true, SUB_FLOW_ID_2);
+    }
+
+    /**
+     * 4 switch ha-flow meter-stats test.
+     * ''''''''/-4--5
+     * 1--2--3
+     * '''''''\----6
+     */
+    @Test
+    public void haFlowMeterStatsTest() throws InterruptedException {
+        HaFlow haFlow = createHaFlow1();
+        sendUpdateHaFlowPathInfo(haFlow);
+        //forward
+        MeterStatsEntry forward = new MeterStatsEntry(METER_ID_HA_FLOW_PATH_FORWARD.getValue(), 400L, 500L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_1, Collections.singletonList(forward)));
+        validateHaFlowMeterStats(HA_FLOW_ID_1, forward, haFlow.getForwardPath().getCookie(), SWITCH_ID_1,
+                METER_ID_HA_FLOW_PATH_FORWARD.getValue(), SUB_FLOW_ID_SHARED, false, true);
+
+        //reverse
+        MeterStatsEntry reverse1 = new MeterStatsEntry(METER_ID_REVERSE_SUB_PATH_1.getValue(), 300L, 400L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_5, Collections.singletonList(reverse1)));
+        validateHaFlowMeterStats(HA_FLOW_ID_1, reverse1, COOKIE_REVERSE_SUBFLOW_1,
+                SWITCH_ID_5, METER_ID_REVERSE_SUB_PATH_1.getValue(), SUB_FLOW_ID_1, false, true);
+
+        MeterStatsEntry reverse2 = new MeterStatsEntry(METER_ID_REVERSE_SUB_PATH_2.getValue(), 200L, 300L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_6, Collections.singletonList(reverse2)));
+        validateHaFlowMeterStats(HA_FLOW_ID_1, reverse2, COOKIE_REVERSE_SUBFLOW_2,
+                SWITCH_ID_6, METER_ID_REVERSE_SUB_PATH_2.getValue(), SUB_FLOW_ID_2, false, true);
+
+        MeterStatsEntry reverseYPoint =
+                new MeterStatsEntry(METER_ID_HA_FLOW_PATH_REVERSE.getValue(), 100L, 200L);
+        sendStatsMessage(new MeterStatsData(SWITCH_ID_3, Collections.singletonList(reverseYPoint)));
+        validateHaFlowMeterStats(HA_FLOW_ID_1, reverseYPoint, COOKIE_REVERSE,
+                SWITCH_ID_3, METER_ID_HA_FLOW_PATH_REVERSE.getValue(),
+                SUB_FLOW_ID_SHARED, true, false);
```

Comment:
`COOKIE_REVERSE` and `SUB_FLOW_ID_SHARED` should be replaced with `null` for `reverseYPoint` because Y point meter has no such tags